antlr/ContactQL.g4

```diff
@@ -5,6 +5,8 @@ import LexUnicode;
 // Lexer rules
 fragment HAS: [Hh][Aa][Ss];
 fragment IS: [Ii][Ss];
+fragment PROPTYPE: (UnicodeLetter)+;
+fragment PROPKEY: (UnicodeLetter | UnicodeDigit | '_')+;
```

Comment:
you could make this stuff stricter but it's actually easier to deal with errors higher up in the parser rather in the lexer